### PR TITLE
Fixes #1292. Fix for Logger Message

### LINKED ISSUE #1292: Issue with MOM6 on macOS: Logger Message
When trying to test out the C12 MOM6 on Parcel (my MacBook), the model crashed with:
```
 Character Resource Parameter: MOM6_GEOSPLUG:libMOM6_GEOSPlug.so
 FormatString.F90         173
 format_vector() - additional unprocessed arguments.
1
```
Working with @tclune we found this was due to a logger message call:

https://github.com/GEOS-ESM/MAPL/blob/f98fae5c158838b991eeda51ebe3e7d93f4d6fea/generic/MAPL_Generic.F90#L4799-L4800

Namely the single quotes. @tclune will look at pFlogger but for now, I'll make a PR for a workaround.

### PR DESCRIPTION
**Pull request by mathomp4**
## Description

Small changes to logger message due to unexpected single quote behavior in pFlogger.

## Related Issue

Closes #1292 

## Motivation and Context

Let's us run MOM6 (really any `AddChildDSO`) on non-.so systems.

## How Has This Been Tested?

I ran on my laptop with Intel. I want to test with GNU as well.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [ ] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [x] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

File: generic/MAPL_Generic.F90 [hunk 1]
```diff
@@ -4796,7 +4796,7 @@ recursive integer function AddChildFromDSO(gc, name, userRoutine, grid, sharedOb
 
       if (.not. is_valid_dso_name(SharedObj)) then
          lgr => logging%get_logger('MAPL.GENERIC')
-         call lgr%warning("AddChildFromDSO: changing shared library extension '%a~' to system specific extension '%a~'.", &
+         call lgr%warning("AddChildFromDSO: changing shared library extension from %a~ to system specific extension %a~", &
               extension, SYSTEM_DSO_EXTENSION)
```
Comment (tclune) [inline generic/MAPL_Generic.F90:4800]:
```suggestion
         call lgr%warning("AddChildFromDSO: changing shared library extension from %a~ to system specific extension %a~", &
              "'"//extension//"'", "'"//SYSTEM_DSO_EXTENSION//"'")
```

**Comment by mathomp4** [comment]
@tclune Well, I get the quotes now:
```
        GENERIC: WARNING: AddChildFromDSO: changing shared library extension from '.so   ' to system specific extension '.dylib'
```
Should we perhaps add a `trim()` to the `extension`? Or maybe we should trim elsewhere?

**Comment by tclune** [comment]
Ah - extensions is bizarrely declared as len=6.  it should be a deferred length string and that will fix the issue.

**Comment by mathomp4** [comment]
> Ah - extensions is bizarrely declared as len=6. it should be a deferred length string and that will fix the issue.

Done. Seems to work.

**Review by atrayano — APPROVED** [review]

**Pull request merged**